### PR TITLE
Ab#51009 Investigate connecting to mpdis from our prototype

### PR DESCRIPTION
**Pull request by TCPrimedPaul**
Pull request of work done for the spike PBI.

### CODE REVIEW

**Comment by AlexandreSanscartier** [comment]
Is it just me or is there no connecting to MPDIS from the current code?

File: CDTS Blazor.sln [hunk 1]
```diff
@@ -6,7 +6,11 @@ Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "CDNApplication", "CDTS Blaz
 EndProject
 Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "CDNApplication.Tests.Integration", "CDNApplication.Tests.Integration\CDNApplication.Tests.Integration.csproj", "{C5E0ECC9-D81B-4074-8DC9-5A26AF502B5F}"
 EndProject
-Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "CDNApplication.Tests.Unit", "CDNApplication.Tests.Unit\CDNApplication.Tests.Unit.csproj", "{40715101-D0E5-48A1-B85B-56B371E4151E}"
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "CDNApplication.Tests.Unit", "CDNApplication.Tests.Unit\CDNApplication.Tests.Unit.csproj", "{40715101-D0E5-48A1-B85B-56B371E4151E}"
+EndProject
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "CDNApplication.WebAPI", "CDNApplication.WebAPI\CDNApplication.WebAPI.csproj", "{6C4414AA-05A0-4617-9E7C-97FBC82B5E60}"
+EndProject
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "SeafarersApiGateway", "..\SeafarersApiGateway_Copy_12_11_On_tctestmaster\SeafarersApiGateway\SeafarersApiGateway.csproj", "{3FCB545F-6130-43DC-B691-ACCFEEA20828}"
```
Comment (AlexandreSanscartier) [inline CDTS Blazor.sln:13]: Which project is this? Just curious.

Comment (tropical-turtle) [reply CDTS Blazor.sln:13]: This is the API Gate way project.
You can test with the following request while staying connected through VPN
https://wwwappstest.tc.gc.ca/Saf-Sec-Sur/4/csf-apim/hello
https://wwwappstest.tc.gc.ca/Saf-Sec-Sur/4/csf-apim/guid

https://wwwappstest.tc.gc.ca/Saf-Sec-Sur/4/csf-apim/pokemon
https://wwwappstest.tc.gc.ca/Saf-Sec-Sur/4/csf-apim/catfacts

the above requests from single point of entry are routed to:

https://tcappstest.tc.gc.ca/saf-sec-sur/4/csf-api/api/hello
https://tcappstest.tc.gc.ca/saf-sec-sur/4/csf-api/api/hello/guid

https://pokeapi.co/api/v2/pokemon
https://cat-fact.herokuapp.com/facts/

**Comment by TCPrimedPaul** [comment]
> I might have missed it, but is there any MPDIS connecting going on in this code?

Not supposed to

**Comment by AlexandreSanscartier** [comment]
> > I might have missed it, but is there any MPDIS connecting going on in this code?
> 
> Not supposed to

It's possible I misunderstood what the spike was about, but I thought it was all about making sure we can connect to MPDIS.

**Comment by TCPrimedPaul** [comment]
> > > I might have missed it, but is there any MPDIS connecting going on in this code?
> > 
> > 
> > Not supposed to
> 
> It's possible I misunderstood what the spike was about, but I thought it was all about making sure we can connect to MPDIS.

The biggest challenge was getting the deployed app communicating with the app hosted on TCAPPS. Since TCAPPS is internal, we already know it can connect to the DB.

**Comment by AlexandreSanscartier** [comment]
> > > > I might have missed it, but is there any MPDIS connecting going on in this code?
> > > 
> > > 
> > > Not supposed to
> > 
> > 
> > It's possible I misunderstood what the spike was about, but I thought it was all about making sure we can connect to MPDIS.
> 
> The biggest challenge was getting the deployed app communicating with the app hosted on TCAPPS. Since TCAPPS is internal, we already know it can connect to the DB.



> > > > I might have missed it, but is there any MPDIS connecting going on in this code?
> > > 
> > > 
> > > Not supposed to
> > 
> > 
> > It's possible I misunderstood what the spike was about, but I thought it was all about making sure we can connect to MPDIS.
> 
> The biggest challenge was getting the deployed app communicating with the app hosted on TCAPPS. Since TCAPPS is internal, we already know it can connect to the DB.

So which part of the code validates that we can contact TCAPPS?

**Comment by TCPrimedPaul** [comment]
If it's not obvious we should ask @tropical-turtle for code walkthrough. My understanding is that the proposed architecture was followed, meaning that a call is made to the API manager, which then relays to API on TCAPPS.

**Comment by AlexandreSanscartier** [comment]
Perfect yeah if we could do a walkthrough this afternoon that would be great. I was trying to see where TCAPPS was getting called and couldn't really figure it out sounds good!

**Review by AlexandreSanscartier — APPROVED** [review]

**Pull request merged**